src/meds_torch/generate_trajectories.py

```diff
@@ -0,0 +1,151 @@
+import torch
+from lightning import LightningDataModule, LightningModule, Trainer
+from lightning.pytorch.loggers import Logger
+from omegaconf import DictConfig
+
+from meds_torch.models import ACTUAL_FUTURE, GENERATE_PREFIX, INPUT_DATA
+from meds_torch.schemas.generate_analysis_schema import (
+    reorder_struct_fields,
+    validate_generated_data,
+)
+from meds_torch.utils import (
+    RankedLogger,
+    extras,
+    instantiate_loggers,
+    log_hyperparameters,
+    task_wrapper,
+)
+from meds_torch.utils.resolvers import setup_resolvers
+
+setup_resolvers()
+log = RankedLogger(__name__, rank_zero_only=True)
+config_yaml = files("meds_torch").joinpath("configs/eval.yaml")
+
+
+def convert_tensors_to_df(batches):
+    keys = ["code", "numeric_value", "time_delta_days", "mask", "numeric_value_mask", "static_mask"]
+    return pl.DataFrame(
+        {
+            key: pl.Series(
+                [
+                    batch[key][i][batch["mask"][i]].tolist()
+                    for batch in batches
+                    for i in range(batch[key].shape[0])
+                ]
+            )
+            for key in keys
+        }
+    )
+
+
+@task_wrapper
+def generate_trajectories(cfg: DictConfig, datamodule=None) -> tuple[dict[str, Any], dict[str, Any]]:
+    """Evaluates given checkpoint on a datamodule testset.
+
+    This method is wrapped in optional @task_wrapper decorator, that controls the
+    behavior during failure. Useful for multiruns, saving info about the crash, etc.
+
+    Args:
+        cfg: DictConfig configuration composed by Hydra.
+    Returns:
+        Tuple[dict, dict] with metrics and dict with all instantiated objects.
+    """
+    assert cfg.ckpt_path
+    if not cfg.data.do_include_subject_id:
+        raise ValueError("Subject ID is required for generating trajectories")
+    if not cfg.data.do_include_prediction_time:
+        raise ValueError("Prediction time is required for generating trajectories")
+
+    log.info(f"Instantiating datamodule <{cfg.data._target_}>")
+    if not datamodule:
+        datamodule: LightningDataModule = hydra.utils.instantiate(cfg.data)
+
+    log.info(f"Instantiating model <{cfg.model._target_}>")
+    model: LightningModule = hydra.utils.instantiate(cfg.model)
+    model.load_state_dict(torch.load(cfg.ckpt_path)["state_dict"])
+
+    log.info("Instantiating loggers...")
+    logger: list[Logger] = instantiate_loggers(cfg.get("logger"))
+
+    log.info(f"Instantiating trainer <{cfg.trainer._target_}>")
+    trainer: Trainer = hydra.utils.instantiate(cfg.trainer, logger=logger)
+
+    object_dict = {
+        "cfg": cfg,
+        "datamodule": datamodule,
+        "model": model,
+        "logger": logger,
+        "trainer": trainer,
+    }
+
+    if logger:
+        log.info("Logging hyperparameters!")
+        log_hyperparameters(object_dict)
+
+    log.info("Starting Generating Predictions!")
+    predictions = trainer.predict(model=model, dataloaders=datamodule)
```

Comment:
This doesn't seem quite right. Let's have a chat about this to make sure I'm understanding this properly as I think I'm confused.